freebridge-ai/routers/analysis.py

```
@@ -206,3 +207,88 @@ async def analyze_general_reputation(request: ReputationAnalysisRequest):
+        {contract_content}
+        </contract_content>
+        """)
+
+        result = await structured_llm.ainvoke(prompt.format(contract_content=truncated_text))
+        return result
+
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.exception("Error parsing and analyzing contract PDF")
+        raise HTTPException(status_code=500, detail="Internal server error")
```

Ruff (0.15.5)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)